src/KokkosComm_request.hpp

```diff
@@ -23,6 +23,22 @@
 namespace KokkosComm {
 
 class Req {
+  // a type-erased callable. Req uses these to attach callbacks to be executed
+  // at wait
+  struct InvokableHolderBase {
+    virtual ~InvokableHolderBase() {}
+
+    virtual void operator()() = 0;
+  };
+  template <Invokable Fn>
+  struct InvokableHolder : InvokableHolderBase {
+    InvokableHolder(const Fn &f) : f_(f) {}
+
+    virtual void operator()() override { return f_(); }
```

Comment:
```suggestion
    virtual void operator()() override { f_(); }
```
? The return type is `void`. Should the return type of `f_` to be `void` be enforced?